### PR TITLE
feat(docs): Add documentation/gh-pages using Catalyst-CI docs builders and actions.

### PR DESCRIPTION
**Pull request by stevenj**
# Description

This is a PR for review only until the catalyst-ci docs builders are merged.

This sets up full documentation building.

Still TODO as a separate PR will be publishing the built docs to github pages.

### CODE REVIEW

**Review by saibatizoku — DISMISSED** [review]
Looks good to me.

File: docs/Earthfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Set the Earthly version to 0.7
+VERSION 0.7
+
+# cspell: words mkdocs runable
+
+# Copy all the source we need to build the docs
+src:
+    FROM github.com/input-output-hk/catalyst-ci/earthly/docs:feat/docs3+mkdocs-material
```
Comment (Mr-Leshiy) [inline docs/Earthfile:8]: It is needed to be referenced to some release tag

File: docs/Earthfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    COPY --dir src includes macros overrides mkdocs.yml /docs
+
+    # Now copy into that any artifacts we pull from the builds.
+    COPY --dir ../+repo-docs/repo /docs/includes
+
+# Build the docs here.
+docs:
+    FROM +src
+
+    DO github.com/input-output-hk/catalyst-ci/earthly/docs:feat/docs3+BUILD
```
Comment (Mr-Leshiy) [inline docs/Earthfile:20]: same

File: docs/Earthfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+# Make a locally runable container that can serve the docs.
+local:
+    # Build a self contained service to show built docs locally.
+    DO github.com/input-output-hk/catalyst-ci/earthly/docs:feat/docs3+PACKAGE
```
Comment (Mr-Leshiy) [inline docs/Earthfile:25]: same

**Review by Mr-Leshiy — CHANGES_REQUESTED** [review]

File: docs/Earthfile (hunk 1 above)
Comment (stevenj) [reply docs/Earthfile]: Yes but the docs in ci are not yet merged, that's why this PR can not merge yet.

**Review by minikin — DISMISSED** [review]
LGTM

**Review by Mr-Leshiy — APPROVED** [review]
LGTM

**Pull request merged**